packages/protocol/contracts/L1/libs/LibProposing.sol

```diff
@@ -105,10 +116,28 @@ library LibProposing {
         view
         returns (TaikoData.Block storage blk)
     {
-        blk = state.blocks[blockId % config.ringBufferSize];
+        blk = state.blocks[blockId % config.blockRingBufferSize];
         if (blk.blockId != blockId) revert L1_BLOCK_ID();
     }
 
+    // If auction is tied to gas, we should charge users based on gas as well. At
+    // this point gasUsed
+    // (in proposeBlock()) is always gasLimit, so use it and in case of
+    // differences refund after verification
+    function getBlockFee(
+        TaikoData.State storage state,
+        TaikoData.Config memory config,
+        uint32 gasLimit
+    )
+        internal
+        view
+        returns (uint64)
+    {
+        // The diff between gasLimit and gasUsed will be redistributed back to
+        // the balance of proposer
+        return state.feePerGas * (gasLimit + config.blockFeeBaseGas);
```

Comment:
Shall we take https://github.com/taikoxyz/taiko-mono/blob/main/packages/protocol/contracts/L2/LibL2Consts.sol#L10 into consideration?

Reply:
Added